Announcing Hanami v2.0.0.alpha4 (#536)
hanami/hanami.github.io: source/blog/2021-12-07-announcing-hanami-200alpha4.html.markdown

---
title: Announcing Hanami v2.0.0.alpha4
date: 2021-12-07 13:15:19 UTC
tags: announcements
author: Luca Guidi
image: true
excerpt: >

>   Streamlined source directories, more flexible settings, better integrated actions, and the beginning of a monthly release cadence.

Reviewer: n.b. I think this is leftover from the last announcement

Author: Fixed it, thanks 😃

---

Hello Hanami community! We're thrilled to announce the release of Hanami 2.0.0.alpha4!

With this new cycle of monthly based releases we have smaller set of changes, but delivered more frequently.

Specifically, we're focusing on the cleanup of our application template.
The template is essential for **you** to try Hanami 2, but also for **us** to shape the ergonomics of the framework.

  * Content Security Policy (new API)
  * Router helpers to be accessible from actions

## Content Security Policy

> We proudly share this story: Hanami was **the first Ruby framework** to ship with Content Security Policy (CSP) support.

Reviewer 2: ```suggestion
We proudly share this story: Hanami was **the first Ruby framework** to ship with [Content Security Policy (CSP)](https://en.wikipedia.org/wiki/Content_Security_Policy) support.
```

In Hanami 1 the public API to handle CSP was less than optimal.
You had to deal with a string blob, with all the complexity of CSP keys and their values.
That was error prone and – frankly – ugly to see.

> Let's start from the last concern: **you won't see CSP setting in newly generated applications**.

Reviewer 2: ```suggestion
Let's start from the last concern: **you won't see CSP setting in newly generated Hanami 2 applications**.
```

> Hanami has now defaults shipped with the framework that doesn't require to be generated with the apps.

Reviewer 2: ```suggestion
Hanami 2 now has a default CSP shipped with the framework, instead of generating it during application creation.
```

First problem solved. 😎

If you need to change a value, or turn off the feature, the [API is simplified](https://github.com/hanami/controller/pull/353).

```ruby
module MyApp
  class Application < Hanami::Application
    # This line will generate the following CSP fragment
    # script-src 'self' https://my.cdn.test;

>     config.actions.content_security_policy[:script_src] += " https://my.cdn.test"

Reviewer 3: Is the leading space intentional?

Author: Yes. 😸
